### ISSUE TITLE
Firebase Auth module has a different structure with Firebase-Admin Auth module

### ISSUE THREAD

**Issue by johnkingzy** [Upstatement/firestore-jest-mock#108]
# Description

You all did a great job with mocking two modules which are
- firebase &
- firebase-admin

KUDOS 👍 🏆 

However, there is an assumption that the Auth module in the firebase module would be the same as the Auth module in firebase-admin  but they are different i.e with different methods and properties

For more clarity: 
This is the [Auth module](https://firebase.google.com/docs/reference/admin/node/admin.auth.Auth-1) for the Firebase Admin
This is the [Auth module](https://firebase.google.com/docs/reference/node/firebase.auth.Auth) for Firebase

if you scroll to methods, you will see that the method list is different.

## Steps to reproduce

Try to use the `auth` module from `firebase-admin`, e.g
```
import * as admin from 'firebase-admin';
admin.initializeApp();

const auth = admin.auth();

auth.createUser({ email: 'test@example.com', password: 'password', displayName: 'john doe' })
```

## Expected result

` TypeError: firebase_1.auth.createUser is not a function`

## Actual result
I can think of two ways this can be resolved
- Have it built-in as part of the `FakeAuth` class
OR
- Create the flexibility to pass it in as a proper in the overrides object

Here is a link to a possible workaround solution
https://gist.github.com/johnkingzy/9097b5b65fc8a472ed51a35d7a0aba45
TLDR: I redefined the `FakeAuth` class to match the module structure for `firebase-admin` 

## Environment
- Local
- Node version: v15.4.0
